pkg/operation/botanist/waiter.go

```diff
@@ -93,38 +93,33 @@ func (b *Botanist) WaitUntilEtcdReady(ctx context.Context) error {
 // WaitUntilKubeAPIServerReady waits until the kube-apiserver pod(s) indicate readiness in their statuses.
 func (b *Botanist) WaitUntilKubeAPIServerReady(ctx context.Context) error {
 	return retry.UntilTimeout(ctx, 5*time.Second, 300*time.Second, func(ctx context.Context) (done bool, err error) {
-		podList := &corev1.PodList{}
-		err = b.K8sSeedClient.Client().List(ctx, podList,
-			client.InNamespace(b.Shoot.SeedNamespace),
-			client.MatchingLabels(map[string]string{"app": "kubernetes", "role": "apiserver"}))
-		if err != nil {
+
+		deploy := &appsv1.Deployment{}
```

Comment:
Did you verify the behavior with the deletion of an API server pod? Is it still responding to requests although it's deleting?

Reply:
I did not verify it with the deletion of the pod. 

What I did is checking all the secrets whether their encryption state matches with the EncryptionConfiguration. Previously I often encountered mismatches (where the EncryptionConfiguration was "active" and the secrets were still not encrypted ... or vice versa). 

After this change in the waiter, I did not encounter any mismatches in my test cluster anymore.